src/Rector/AbstractRector.php

```diff
@@ -263,7 +264,14 @@ protected function mirrorComments(Node $newNode, Node $oldNode): void
             return;
         }
 
-        $newNode->setAttribute(AttributeKey::PHP_DOC_INFO, $oldNode->getAttribute(AttributeKey::PHP_DOC_INFO));
+        $oldPhpDocInfo = $oldNode->getAttribute(AttributeKey::PHP_DOC_INFO);
+        $newPhpDocInfo = $newNode->getAttribute(AttributeKey::PHP_DOC_INFO);
+
+        if ($newPhpDocInfo instanceof PhpDocInfo && $oldPhpDocInfo !== $newPhpDocInfo) {
+            return;
+        }
```

Comment:
PhpDocInfo can be grabbed via `createFromNodeOrEmpty()`, so it can be empty PhpDocNode, and then it reduplicated by new node, also, comparing new re-create object will cause false, see https://3v4l.org/Y9qSj

so, compare the string result instead:

```suggestion
        $newPhpDocInfo = $newNode->getAttribute(AttributeKey::PHP_DOC_INFO);
        
        if ($newPhpDocInfo instanceof PhpDocInfo && $newPhpDocInfo->getPhpDocNode()->children !== []) {
            $oldPhpDocInfo = $oldNode->getAttribute(AttributeKey::PHP_DOC_INFO);
            if (! $oldPhpDocInfo instanceof PhpDocInfo) {
                 return;
            }
            
            if ((string) $oldPhpDocInfo->getPhpDocNode() !== (string) $newPhpDocInfo->getPhpDocNode()) {
                 return;
            }
        }
```

Reply:
The 3v4l link does not seem to be correct, but I get what you mean, updated